.github/workflows/dependency-zip.yml

```diff
@@ -0,0 +1,166 @@
+        required: true
+      S3_ENDPOINT:
+        description: "S3 endpoint"
+        required: true
+    outputs:
+      use-release:
+        description: "Whether to use the release zip"
+        value: ${{ jobs.check-zip.outputs.use-release }}
+      s3-zip-name:
+        description: "Name of the dependency zip file in S3"
+        value: ${{ jobs.check-zip.outputs.s3-zip-name }}
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+
+    outputs:
+      checkout-path: ${{ steps.checkout_path.outputs.path }}
+      head-branch: ${{ steps.branch_values.outputs.head_branch }}
+      base-branch: ${{ steps.branch_values.outputs.base_branch }}
+      is-main: ${{ steps.branch_values.outputs.head_branch == 'main' }}
+
+    steps:
+      - name: Define the checkout path
+        id: checkout_path
+        run: |
+          repository=${{ inputs.repository }}
+          echo "path=${repository//\//_}" >> $GITHUB_OUTPUT
+
+      - name: Define the head and base branches
+        id: branch_values
+        run: |
+          if [ -z "$GITHUB_HEAD_REF" ]; then
+            echo "head_branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
+          else
+            echo "head_branch=$GITHUB_HEAD_REF" >> $GITHUB_OUTPUT
+          fi
+
+          if [ -z "$GITHUB_BASE_REF" ]; then
+            echo "base_branch=main" >> $GITHUB_OUTPUT
+          else
+            echo "base_branch=$GITHUB_BASE_REF" >> $GITHUB_OUTPUT
+          fi
+
+  check-branch:
+    runs-on: ubuntu-latest
+
+    needs:
+      - prepare
+
+    # If the head branch is main, we don't need to check the dependency branch, as we will use the release zip.
+    if: ${{ needs.prepare.outputs.is-main == 'false' }}
+
+    outputs:
+      branch-to-checkout: ${{ steps.output_values.outputs.dependency_branch }}
+      is-main: ${{ steps.output_values.outputs.dependency_branch == 'main' }}
+
+    steps:
+      - name: Checkout dependency to verify the available branches
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ inputs.repository }}
+          ref: main
+          path: ${{ needs.prepare.outputs.checkout-path }}
+          fetch-depth: 1
+          token: ${{ secrets.GITHUB_CHECKOUT_TOKEN }}
+
+      - name: Check if the required branch exists
+        id: output_values
+        run: |
+          cd ${{ needs.prepare.outputs.checkout-path }}
+
+          head_branch_exists=$(git ls-remote --heads origin ${{ needs.prepare.outputs.head-branch }} | wc -l)
+
+          if [ $head_branch_exists -eq 1 ]; then
+            echo "dependency_branch=${{ needs.prepare.outputs.head-branch }}" >> $GITHUB_OUTPUT
+          else
+            base_branch_exists=$(git ls-remote --heads origin ${{ needs.prepare.outputs.base-branch }} | wc -l)
+
+            if [ $base_branch_exists -eq 1 ]; then
+              echo "dependency_branch=${{ needs.prepare.outputs.base-branch }}" >> $GITHUB_OUTPUT
+            else
+              echo "dependency_branch=main" >> $GITHUB_OUTPUT
```

Comment:
Let's change main to a variable that can be passed in with main as the default. Some repos still have a base of master

Reply:
Done!